framework/src/postprocessors/PseudoTimestep.C

```diff
@@ -0,0 +1,213 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// Navier-Stokes includes
+#include "PseudoTimestep.h"
+#include "FEProblemBase.h"
+#include "NonlinearSystemBase.h"
+#include "Transient.h"
+#include "Restartable.h"
+
+// MOOSE includes
+#include "libmesh/quadrature.h"
+#include "libmesh/enum_norm_type.h"
+#include "metaphysicl/raw_type.h"
+
+#include <algorithm>
+#include <limits>
+
+registerMooseObject("MooseApp", PseudoTimestep);
+
+InputParameters
+PseudoTimestep::validParams()
+{
+  InputParameters params = GeneralPostprocessor::validParams();
+
+  /// Enum that is used to select the timestep-selection method.
+  MooseEnum method("SER=1 EXP=2 RDM=3");
+  params.addParam<MooseEnum>("method", method, "The method used for CFL scaling");
+  params.addRequiredParam<Real>("initial_dt", "Initial timestep");
+  params.addRequiredParam<Real>("alpha", "The parameter alpha used in the scaling of the timestep");
+  params.addParam<Real>("max_dt", "The largest timestep allowed");
+  params.addParam<unsigned int>(
+      "iterations_window",
+      1.0,
+      "For how many iterations should the CFL be tracked (only applies to the SER method)");
+  // Because this post-processor is meant to be used with PostprocessorDT, it
+  // should be executed on initial (not included by default) and timestep end.
+  params.set<ExecFlagEnum>("execute_on") = {EXEC_TIMESTEP_END, EXEC_INITIAL};
+  params.suppressParameter<ExecFlagEnum>("execute_on");
+
+  params.addClassDescription("Computes a time step size to accelerate iterations");
```

Comment:
there's no universal accelerator for every equation afaik. If it s literally proven to work for NS (and you dont want people to use it as a heuristic for other equations) then maybe we should move this to the NS module? the CFL criterion postprocessor lives there

Reply:
The methodology is universal and provides guidelines for how to devise a pseudo transient continuation algorithm and shows it converges. https://apps.dtic.mil/sti/tr/pdf/ADA314613.pdf  The methods implemented are some of the main methods derived from this and used in applications, and the applications tend to be CFD cause they are the most expensive ones, but there's no reason it has to be only CFD. 
Heuristic means experimental, which implies there's no umbrella theory for it, which is not the case here.

Reply:
And btw CFL is not only for CFD either.

Reply:
ok so SER works has a proof for a category of equations in that paper you linked. For those equations I agree it's not a heuristic to use it. Do RDM and EXP have a similar proof?
I would argue switching out a steady-state solve to a transient-to-steady-state is in itself a heuristic. It's good for some problems. It's a terrible idea for others (neutronics for example)

Reply:
is there a word that fits better between "it will accelerate" (which is not true for all problems) and "heuristic" (which has too much experimental / trial-and-error connotation seemingly here)

Reply:
I take Peter's change to speed this up. Side comment: Steady state is by definition when du/dt=0, so a subset of transients. If time dependency is heuristic then life on earth might as well not exist.